frontend/src/components/assessments/student-experience/Question.tsx

```diff
@@ -18,53 +21,64 @@ import QuestionText from "./question-components/QuestionText";
 import ShortAnswer from "./question-components/ShortAnswer";
 import Text from "./question-components/Text";
 
-interface DisplayQuestionProps {
-  questionComponents: QuestionComponentResponse[];
+interface QuestionProps {
+  elements: QuestionComponentResponse[];
 }
 
-const Question = ({
-  questionComponents,
-}: DisplayQuestionProps): React.ReactElement => {
+const Question = ({ elements }: QuestionProps): React.ReactElement => {
+  let counter = -1;
+  const answerElementIndex = elements.map((element) => {
+    if (element.type in ResponseElementType) {
+      counter += 1;
+      return counter;
+    }
+    return 0;
+  });
```

Comment:
i really don't like this, but i'm not sure what a better way to do this is!

Reply:
I guess it might be a little clearer if you rename `counter` -> `answersSoFar`. You could also do something like `elements.map((element) => element.type in ResponseElementType ? ++answersSoFar : 0)` if you wanted to have a shorter line, but I feel like that's actually less readable. Probably most explicit would be something like:

```tsx
const answersSoFar = 0;
const answerElementIndex: { [key: number]: number } = {};
elements.forEach(({ type }, index) => {
  if (type in ResponseElementType) {
    answerElementIndex[index] = answersSoFar;
    ++answersSoFar;
  }
});
```

You'd probably want to memoize this at any rate (you can use it for the dict I mentioned before!)

Reply:
i actually quite like your one liner! i updated to that, but let me know if you hate it & i can refactor. +1 to memoize